### PR TITLE
[Merged by Bors] - v2alpha1: Implement v2alpha1 ActivationService.Highest rpc

### LINKED ISSUE #6539: v2alpha1: Add Highest rpc to Activation service
## Description

V2 API lacks the Highest ATX method, which is present in V1.

In V1, the Highest method is defined as follows:
```
message HighestResponse {
    Activation atx = 1;
}

// Highest returns the atx id with the highest tick count.
rpc Highest(google.protobuf.Empty) returns (HighestResponse);
```

### PR DESCRIPTION
**Pull request by kacpersaw**
## Description
Closes #6539

This pull request introduces several significant changes to the `api/grpcserver/v2alpha1/activation.go` file, including the addition of a new interface, modifications to the `ActivationService` struct, and the implementation of new methods. Additionally, it includes updates to the testing framework and dependencies. Below is a summary of the most important changes.

### Changes to `activation.go`:

* Added a new `activationProvider` interface with methods `GetAtx` and `MaxHeightAtx`.
* Modified the `NewActivationService` function to accept additional parameters `atxProvider` and `goldenAtx`.
* Added a new method `Highest` to the `ActivationService` struct to return the highest ATX.

### Mock and Test Updates:

* Generated a mock for the `activationProvider` interface in `activation_mocks.go`.
* Updated tests in `activation_test.go` to use the new `activationProvider` mock and added a new test for the `Highest` method. [[1]](diffhunk://#diff-da2bd880d3c3e3214a6eb2737b9f4c53ad1a803ea6cd93f6735d97d29d2ba04eR29-R30) [[2]](diffhunk://#diff-da2bd880d3c3e3214a6eb2737b9f4c53ad1a803ea6cd93f6735d97d29d2ba04eL36-R41) [[3]](diffhunk://#diff-da2bd880d3c3e3214a6eb2737b9f4c53ad1a803ea6cd93f6735d97d29d2ba04eR235-R237) [[4]](diffhunk://#diff-da2bd880d3c3e3214a6eb2737b9f4c53ad1a803ea6cd93f6735d97d29d2ba04eL247-R256) [[5]](diffhunk://#diff-da2bd880d3c3e3214a6eb2737b9f4c53ad1a803ea6cd93f6735d97d29d2ba04eR287-R339)

### CODE REVIEW

File: node/node.go [hunk 1]
```diff
@@ -1554,7 +1554,7 @@ func (app *App) grpcService(svc grpcserver.Service, lg log.Log) (grpcserver.Serv
 		app.grpcServices[svc] = service
 		return service, nil
 	case v2alpha1.Activation:
-		service := v2alpha1.NewActivationService(app.apiDB)
+		service := v2alpha1.NewActivationService(app.apiDB, app.cachedDB, types.ATXID(app.Config.Genesis.GoldenATX()))
```
Comment (fasmat) [inline node/node.go]: I thought `app.cachedDB` should not be used in the API any more? The method `MaxHeightAtx` is just `atxs.GetIDWithMaxHeight(db, types.EmptyNodeID, atxs.FilterAll)` and `GetAtx` is also just `atxs.Get(db, id)`.

File: api/grpcserver/v2alpha1/activation.go [hunk 2]
```diff
@@ -28,6 +29,11 @@ const (
 	ActivationStream = "activation_stream_v2alpha1"
 )
 
+type activationProvider interface {
+	GetAtx(id types.ATXID) (*types.ActivationTx, error)
+	MaxHeightAtx() (types.ATXID, error)
+}
```
Comment (fasmat) [inline api/grpcserver/v2alpha1/activation.go]: This should not be necessary.

**Review by fasmat — CHANGES_REQUESTED** [review]

**Review by fasmat — APPROVED** [review]

**Comment by kacpersaw** [comment]
bors merge

**Comment by fasmat** [comment]
bors merge

**Pull request closed without merging**